core/src/main/java/org/testcontainers/dockerclient/DockerClientProviderStrategy.java

```diff
@@ -39,6 +43,14 @@
      */
     public abstract String getDescription();
 
+    protected boolean isApplicable() {
+        return true;
+    }
+
+    protected int getPriority() {
```

Comment:
*Maybe* worth mentioning in a comment how the priority is used (i.e. highest to lowest) for avoidance of doubt.